strax/processing/hitlets.py

```diff
@@ -0,0 +1,526 @@
+import numpy as np
+import numba
+
+import strax
+from strax import utils
+from strax.dtypes import hit_dtype
+export, __all__ = strax.exporter()
+
+# ----------------------
+# Hitlet building:
+# ----------------------
+@export
+def concat_overlapping_hits(hits, extensions, pmt_channels):
+    """
+    Function which concatenates hits which may overlap after left and 
+    right hit extension. Assumes that hits are sorted correctly.
+    
+    Note: 
+        This function only updates time, length and record_i of the hit.
+        (record_i is set according to the first hit)
+    
+    :param hits: Hits in records.
+    :param extensions: Tuple of the left and right hit extension.
+    :param pmt_channels: Tuple of the detectors first and last PMT
+    """
+    # Getting channel map and compute the number of channels:
+    first_channel, last_channel = pmt_channels
+    nchannels = last_channel - first_channel + 1
+
+    # Buffer for concat_overlapping_hits, if specified in 
+    # _concat_overlapping_hits numba crashes.
+    last_hit_in_channel = np.zeros(nchannels,
+                                   dtype=(hit_dtype
+                                          + [(('End time of the interval (ns since unix epoch)',
+                                               'endtime'), np.int64)]))
+
+    if len(hits):
+        hits = _concat_overlapping_hits(hits, extensions, first_channel, last_hit_in_channel)
+    return hits
+
+
+@utils.growing_result(strax.hit_dtype, chunk_size=int(1e4))
+@numba.njit(nogil=True, cache=True)
+def _concat_overlapping_hits(hits,
+                             extensions,
+                             first_channel,
+                             last_hit_in_channel,
+                             _result_buffer=None):
+    buffer = _result_buffer
+    offset = 0
+
+    le, re = extensions
+    dt = hits['dt'][0]
+    assert np.all(hits['dt'] == dt), 'All hits must have the same dt!'
+
+    for h in hits:
+        st = h['time'] - int(le * h['dt'])
```

Comment:
I think theoretically yes, you can get this edge case if you only apply this function to the data. But only if the chunk start is equivalent with the start of a record and you find a hit within the very first sample of this record. And this should basically never happen. 
Further get_hitlets_data, will correct the start time in case a hit is extended into a not recorded region (same for et).

Reply:
> And this should basically never happen.

why not? Also, get_hitlets_data doesn't do endtimes right

Reply:
Because the digitizers and hitfinder should share the same thresholds. Otherwise Acceptance computation becomes odd. get_hitlet_data will change the `length` field according to the length of the found data.

I do not see any feasible way to change this.

Reply:
Never mind I just had an idea. I could require to pass also the chunk start and end time as additional arguments to the function.